src/code/z_actor.c

```diff
@@ -3400,110 +3447,156 @@ s32 func_800BB59C(PlayState* play, Actor* actor) {
     return (x > -20) && (x < gScreenWidth + 20) && (y > -160) && (y < gScreenHeight + 160);
 }
 
-void func_800BB604(GameState* gameState, ActorContext* actorCtx, Player* player, s32 actorCategory) {
-    PlayState* play = (PlayState*)gameState;
-    f32 temp_f0_2;
-    Actor* targetedActor;
-    Actor* actor;
-    s32 phi_s2;
-    CollisionPoly* sp80;
-    s32 sp7C;
-    Vec3f sp70;
+/**
+ * Search for targetable actors of the `actorCategory` category.
+ *
+ * Looks for the actor of said category with higher targetPriority and the one that is nearest to player. This actor
+ * must be within the range (relative to player) speicified by its targetMode.
+ *
+ * The actor must be on-screen (more or less, not sure what's going on in Target_InTargetableScreenRegion)
```

Comment:
```suggestion
 * The actor must be on-screen
```
Are you still unsure of `Target_InTargetableScreenRegion`? It's just a region that extends outside the screen